Python/pylifecycle.c

```diff
@@ -2121,6 +2132,14 @@ new_interpreter(PyThreadState **tstate_p, const PyInterpreterConfig *config)
         goto error;
     }
 
+    // initialize the interp->obmalloc state.  This must be done after
+    // the settings are loaded (so that feature_flags are set) but before
+    // any calls are made to obmalloc functions.
+    if (!_PyMem_init_obmalloc(interp, runtime)) {
+        status = _PyStatus_NO_MEMORY();
+        goto error;
+    }
```

Comment:
Likewise, if we did not make `PyInterpreterState.obmalloc` a pointer and kept the existing static initializer code, then we wouldn't need to do this.

Reply:
Then again, it's basically the same as what we were doing in `init_interpreter()`.  At the same time, it seemed a bit more natural where it was in `init_interpreter()` and seems more out of place here.  Note that we've been making some effort over the last few years to make the different steps in initialization less coupled and more compartmentalized.

You're right about needing the config though.  Perhaps we should be passing the config to `PyInterpreterState_New()` and move the `init_interp_settings()` and `_PyMem_init_obmalloc()` calls down.

Regardless, if there isn't a simple obvious solution on this we can punt to later.